rollup.config.js

```diff
@@ -32,6 +33,18 @@ function resolveFileUrl({ fileName }) {
   return JSON.stringify(fileName.replace(/^static\//, '/'));
 }
 
+const dir = '.tmp/build';
+const staticPath = 'static/[name]-[hash][extname]';
+const jsPath = staticPath.replace('[extname]', '.js');
+
+function jsFileName(chunkInfo) {
+  const parsedPath = parsePath(chunkInfo.facadeModuleId);
+  if (parsedPath.name !== 'index') return jsPath;
+  // Come up with a better name than 'index'
+  const name = parsedPath.dir.split('/').slice(-1);
```

Comment:
Need to protect against this being the cwd:

```suggestion
  const name = parsedPath.dir.split('/').slice(-1) || 'index';
```

Reply:
I'm not sure this is needed. The path is the full path on disk, so it shouldn't be falsey. For scripts that are in one of the roots, the name will end up as "client" or "shared", which seems ok.

Reply:
ah good point, as long as its not the parent directory name then it's at least deterministic.